src/sql/parts/grid/views/editData/editData.component.ts

```diff
@@ -245,7 +248,14 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
 			// If the user is deleting a new row that hasn't been committed yet then use the revert code
 			if (self.newRowVisible && index === self.dataSet.dataRows.getLength() - 2) {
 				self.revertCurrentRow();
-			} else {
+			}
+			else if (self.isNullRow(index)) {
+				// Don't try to delete NULL (new) row since it doesn't actually exist and will throw an error
+				// TODO : We should really just stop the context menu from showing up for this row, but that's a bit more involved
```

Comment:
I'd suggest adding an issue number or at least this PR number in this comment so we can refer back to it in the future